rest-api/dao/physical_measurements_dao.py

```diff
@@ -553,3 +597,75 @@ def from_client_json(resource_json, participant_id=None, **unused_kwargs):
                                 createdUsername=created_username, finalizedSiteId=finalized_site_id,
                                 finalizedUsername=finalized_username)
 
+  @staticmethod
+  def _validate_patch_update(measurement, resource):
+    """validates request of resource"""
+    cancelled_required_fields = ['status', 'reason', 'cancelledInfo']
+    restored_required_fields = ['status', 'reason', 'restoredInfo']
+
+    if resource['status'].lower() == 'cancelled':
```

Comment:
If status is missing this will give a KeyError. I think you need to use get() here  and check for status first?